### PR TITLE
First setup

### PR DESCRIPTION
**Pull request by tcarmet**

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+name: 'Artifacts Setup'
+description: 'Set outputs to work with Artifacts Actions'
+inputs:
+  artifacts_url:
+    description: 'The url of Artifacts'
+    required: false
+    # TODO: discuss the fact that secrets cannot be set
+    # as defaults on inputs.
+    default: 'https://artifacts.scality.net'
```
Comment (erwan-b) [inline action.yaml]: As this thing is in public, remove unwanted default value than return to scality ?

File: action.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Setup new properties for artifacts name
+      id: properties
+      run: |
+        echo "::set-output name=repository::$(echo ${{ github.repository }} | sed 's/\//:/')"
+        echo "::set-output name=commit-timestamp::$(date --date='${{ github.event.head_commit.timestamp }}' +"%s")"
+        echo "::set-output name=commit-short-revision::$(echo -n '${{ github.sha }}' | cut -c1-10)"
+        echo "::set-output name=workflow-name::$(echo -n '${{ github.workflow }}' | sed 's/\W/-/g' | sed 's/^-//g')"
```
Comment (erwan-b) [inline action.yaml:28]: That's great

**Review by erwan-b — APPROVED** [review]

File: action.yaml (hunk 1 above)
Comment (tcarmet) [reply action.yaml]: We need to discuss this one together, you are right, I was hoping we could find a way to define those as the following:
```suggestion
    default: '${{ secret.ARTIFACTS_URL }}'
```

But secrets are not supported as default value on the input :cry: 

So I think we need to define this as "required" with no default value and make the user set the secret on the other end.

Comment (tcarmet) [inline action.yaml]:
```suggestion
    description: 'The url of Artifacts'
    required: true
```

File: action.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+---
+name: 'Artifacts Setup'
+description: 'Set outputs to work with Artifacts Actions'
+inputs:
+  artifacts_url:
```
Comment (tcarmet) [inline action.yaml]:
```suggestion
  artifacts-url:
```

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+on: push
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Artifacts Setup
+        id: artifacts-setup
+        uses: ./
```
Comment (tcarmet) [inline .github/workflows/test.yaml:13]:
```suggestion
      - name: Artifacts Setup
        id: artifacts-setup
        uses: ./
        with:
          artifacts-url: ${{ secrets.ARTIFACTS_URL }}
```

**Pull request merged**